pkg/controller/gitea/gitea_controller.go

```diff
@@ -0,0 +1,157 @@
+	if err != nil {
+		return err
+	}
+
+	// TODO(user): Modify this to be the types you create that are owned by the primary resource
+	// Watch for changes to secondary resource Pods and requeue the owner Gitea
+	err = c.Watch(&source.Kind{Type: &corev1.Pod{}}, &handler.EnqueueRequestForOwner{
+		IsController: true,
+		OwnerType:    &integreatlyv1alpha1.Gitea{},
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileGitea{}
+
+// ReconcileGitea reconciles a Gitea object
+type ReconcileGitea struct {
+	// This client, initialized using mgr.Client() above, is a split client
+	// that reads objects from the cache and writes to the apiserver
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// Reconcile reads that state of the cluster for a Gitea object and makes changes based on the state read
+// and what is in the Gitea.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileGitea) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	log.Printf("Reconciling Gitea %s/%s\n", request.Namespace, request.Name)
+
+	// Fetch the Gitea instance
+	instance := &integreatlyv1alpha1.Gitea{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+
+	instanceCopy := instance.DeepCopy()
+
+	// Try create all gitea resources
+	r.CreateResource(instanceCopy, GiteaServiceAccountName)
+	r.CreateResource(instanceCopy, GiteaPgServiceName)
+	r.CreateResource(instanceCopy, GiteaPgDeploymentName)
+	r.CreateResource(instanceCopy, GiteaServiceName)
+	r.CreateResource(instanceCopy, GiteaDeploymentName)
+	r.CreateResource(instanceCopy, GiteaReposPvcName)
+	r.CreateResource(instanceCopy, GiteaPgPvcName)
+	r.CreateResource(instanceCopy, GiteaConfigMapName)
+
+	// The oauth-proxy is only compatible with Openshift because it
+	// does not support ingress
+	if instanceCopy.Spec.DeployProxy {
+		r.CreateResource(instanceCopy, ProxyServiceAccountName)
+		r.CreateResource(instanceCopy, ProxyServiceName)
+		r.CreateResource(instanceCopy, ProxyDeploymentName)
+		r.CreateResource(instanceCopy, ProxyRouteName)
+	} else {
+		r.CreateResource(instanceCopy, GiteaIngressName)
+	}
+
+	return reconcile.Result{}, nil
+}
+
+// Creates a generic kubernetes resource from a template
+func (r *ReconcileGitea) CreateResource(cr *integreatlyv1alpha1.Gitea, resourceName string) {
```

Comment:
is it expected that we will just log errors here?

Reply:
I could return the error in the `Reconcile` function to reque the request (and as far as i understand it, rerun it immediately) but i'm not sure if this would be helpful: if it failed the first time it would likely fail again. Also, even though i return no error in the `Reconcile` function, it is still invoked again after a while (and resource creation will be attempted again).